packages/eslint-plugin-reselect-utils/src/utils/arePropsDifferent.ts

```diff
@@ -0,0 +1,38 @@
+import * as ts from 'typescript';
+import { getTypeFromSymbol } from './getTypeFromSymbol';
+
+export const arePropsDifferent = (
+  selectorProperties: ts.Symbol[],
+  keySelectorProperties: ts.Symbol[],
+  typeChecker: ts.TypeChecker,
+) => {
+  if (selectorProperties.length !== keySelectorProperties.length) {
+    return true;
+  }
+
+  return selectorProperties.some((selectorProperty) => {
+    const {
+      type: propertyType,
+      isOptional: isPropertyOptional,
+    } = getTypeFromSymbol(selectorProperty, typeChecker);
+    const propertyTypeString = propertyType
+      ? typeChecker.typeToString(propertyType)
+      : 'any';
+
+    return !keySelectorProperties.find((keySelectorProperty) => {
+      const {
+        type: keySelectorPropertyType,
+        isOptional: isKeySelectorPropertyOptional,
+      } = getTypeFromSymbol(keySelectorProperty, typeChecker);
+      const keySelectorPropertyTypeString = keySelectorPropertyType
+        ? typeChecker.typeToString(keySelectorPropertyType)
+        : 'any';
```

Comment:
may be it would be safer to suggest that if we cant infer types that means that is different types, for example 'unknownPropsType' and 'unknownKeysSelectorPropsType'

Reply:
Added unknownPropsType